### PR TITLE
Add support for image and image stream entities

### PR DESCRIPTION
**Pull request by zakiva**

### CODE REVIEW

**Comment by zakiva** [comment]
@simon3z Please review

**Comment by simon3z** [comment]
:+1:

**Comment by moolitayer** [comment]
LGTM :+1:

**Comment by simon3z** [comment]
@zakiva @moolitayer remember that you have to try and push annotations to the image. Look into `PATCH`. I think you'll have to add the support into kubeclient.

**Comment by simon3z** [comment]
@abonas can you review?

File: test/json/images_list.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            "io.openshift.builder-base-version": "f5bfa20",
+            "io.openshift.builder-version": "33dadac",
+            "io.openshift.expose-services": "8080:http",
+            "io.openshift.s2i.scripts-url": "image:///usr/libexec/s2i",
+            "io.openshift.tags": "builder,ruby,ruby20",
+            "io.s2i.scripts-url": "image:///usr/libexec/s2i"
+          }
+        },
+        "Architecture": "amd64",
+        "Size": 53514887
```
Comment (abonas) [inline test/json/images_list.json:66]: given that this is json, is this ok that this is a number and not a string representing a number?

**Comment by abonas** [comment]
@simon3z I remember there were some entities like ImageStream, etc - how do they relate to the Image entity? (and are they needed as well?)

**Comment by simon3z** [comment]
> @simon3z I remember there were some entities like ImageStream, etc - how do they relate to the Image entity? (and are they needed as well?)

ImageStream and Image are very similar entities. The simple explanation is that the ImageStream contains information for the user/developer, on the other hand Image contains information for the operator (kubelet, etc.).

@zakiva since we're at it can you add `ImageStream` as well? Thanks.

File: test/json/images_list.json (hunk 1 above)
Comment (simon3z) [inline test/json/images_list.json:66]: @abonas this is a good question for openshift (issue, etc.).

Comment (moolitayer) [inline test/json/images_list.json:66]: afaik numbers (unlike strings) can be bare http://json.org/

**Comment by zakiva** [comment]
> @zakiva since we're at it can you add ImageStream as well? Thanks.

@simon3z added
cc @abonas

File: test/json/image_stream.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "eng-boss-image-inspector",
+    "namespace": "default",
+    "selfLink": "/oapi/v1/namespaces/default/imagestreams/eng-boss-image-inspector",
+    "uid": "f4499f16-b7bc-11e5-a0a4-001a4a231290",
+    "resourceVersion": "2836367",
+    "creationTimestamp": "2016-01-10T17:09:48Z",
+    "annotations": {
+      "openshift.io/image.dockerRepositoryCheck": "the registry \"https://rcm-img-docker01.build.eng.bos.redhat.com:5001\" could not be reached"
```
Comment (abonas) [inline test/json/image_stream.json:12]: kinda weird annotation, isn't it? @smarterclayton, any idea why this is an annotation and not a status/err message?

**Comment by abonas** [comment]
other than [that](https://github.com/abonas/openshift_client/pull/22#discussion_r57527053), @simon3z lgtm

File: test/json/images_list.json (hunk 1 above)
Comment (simon3z) [inline test/json/images_list.json:66]: @moolitayer yes they can be bare, but IIRC there was something about 64bit integers not supported everywhere so in general they're passed as strings.

**Pull request merged**